BlueShift-iOS-SDK/InApps/BlueShiftNotificationViewController.m

```diff
@@ -120,12 +121,25 @@ - (void)setBackgroundImageFromURL:(UIView *)notificationView {
     if (notificationView && self.notification.templateStyle && self.notification.templateStyle.backgroundImage &&
     ![self.notification.templateStyle.backgroundImage isEqualToString:@""]) {
         NSString *backgroundImageURL = self.notification.templateStyle.backgroundImage;
-        NSData * imageData = [[NSData alloc] initWithContentsOfURL: [NSURL URLWithString: backgroundImageURL]];
-        UIImage *image = [[UIImage alloc] initWithData:imageData];
-        [notificationView setBackgroundColor: [UIColor colorWithPatternImage: image]];
+        UIImage *image = [[UIImage alloc] initWithData:[self loadAndCacheImageForURLString:backgroundImageURL]];
+        UIImageView *imageView = [[UIImageView alloc] initWithImage:image];
+        imageView.frame = notificationView.bounds;
+        imageView.contentMode = UIViewContentModeScaleAspectFill;
+        [notificationView addSubview:imageView];
     }
 }
 
+-(NSData*)loadAndCacheImageForURLString:(NSString*)urlString {
+    NSData *imageData = [[NSData alloc] init];
+    if([cachedImageData valueForKey: urlString]) {
+        imageData = [cachedImageData valueForKey:urlString];
+    } else {
+        imageData = [[NSData alloc] initWithContentsOfURL: [NSURL URLWithString: urlString]];
```

Comment:
should we also consider storing this file locally? to avoid a network call on app restart?

Reply:
There is very little chance of same image getting used for other in-app messages, so storing image locally will not be good option.

Reply:
Yes, but this method is called only when you are setting the image, so ideally the cache will not be used right? always this will be pulled from the url